### PR TITLE
Add LS extension endpoint to retrieve type information of params and return type desc of a given function definition

### PR DESCRIPTION
**Pull request by madushajg**
## Purpose
$Subject

Fixes https://github.com/ballerina-platform/ballerina-lang/issues/38612

## Check List 
- [ ] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/extensions/ballerina/symbol/BallerinaSymbolService.java [hunk 1]
```diff
@@ -220,6 +222,48 @@ public CompletableFuture<TypesFromSymbolResponse> getTypeFromSymbol(TypeFromSymb
...
+            Optional<Path> filePath = PathUtil.getPathFromURI(fileUri);
+            if (filePath.isEmpty()) {
+                return typeFromSymbolResponse;
+            }
+            List<ResolvedTypeForSymbol> types = new ArrayList<>();
+            try {
+                Optional<SemanticModel> semanticModel =
+                        this.workspaceManagerProxy.get(fileUri).semanticModel(filePath.get());
+                Optional<Document> document = workspaceManagerProxy.get(fileUri).document(filePath.get());
+                if (semanticModel.isPresent() && document.isPresent()) {
```
Comment (malinthar) [inline language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/extensions/ballerina/symbol/BallerinaSymbolService.java:239]: Shall we `early return` here checking for absence

File: language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/extensions/ballerina/symbol/BallerinaSymbolService.java [hunk 2]
```diff
@@ -220,6 +222,48 @@ public CompletableFuture<TypesFromSymbolResponse> getTypeFromSymbol(TypeFromSymb
...
+                    LinePosition fnPosition = request.getFnPosition();
+                    Optional<Symbol> symbol = semanticModel.get().symbol(document.get(), fnPosition);
+                    if (symbol.isPresent()) {
```
Comment (malinthar) [inline language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/extensions/ballerina/symbol/BallerinaSymbolService.java:242]: ditto

**Review by malinthar — COMMENTED** [review]
LGTM! Add some minor comments

File: language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/extensions/ballerina/symbol/BallerinaSymbolService.java (hunk 2 above)
Comment (madushajg) [reply language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/extensions/ballerina/symbol/BallerinaSymbolService.java]: Addressed with 7b7dc194b28a59dbbdc3791f4b3d4f4b1bbfb74e

File: language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/extensions/ballerina/symbol/BallerinaSymbolService.java (hunk 1 above)
Comment (madushajg) [reply language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/extensions/ballerina/symbol/BallerinaSymbolService.java]: Addressed with 7b7dc194b28a59dbbdc3791f4b3d4f4b1bbfb74e

**Review by malinthar — APPROVED** [review]

**Pull request merged**